src/tsickle.ts

```diff
@@ -867,15 +867,31 @@ class Annotator extends ClosureRewriter {
     }
   }
 
-  private visitProperty(namespace: string[], p: ts.Declaration) {
-    let name = this.propertyName(p);
+  /**
+   * @param optional If true, property is optional (e.g. written "foo?: string").
+   */
+  private visitProperty(namespace: string[], prop: ts.Declaration, optional = false) {
+    let name = this.propertyName(prop);
     if (!name) {
-      this.emit(`/* TODO: handle strange member:\n${this.escapeForComment(p.getText())}\n*/\n`);
+      this.emit(`/* TODO: handle strange member:\n${this.escapeForComment(prop.getText())}\n*/\n`);
       return;
     }
 
-    let tags = this.getJSDoc(p) || [];
-    tags.push({tagName: 'type', type: this.typeToClosure(p)});
+    let type = this.typeToClosure(prop);
+    // Interesting special case:
+    // When a property is optional, e.g.
+    //   foo?: string;
+    // Then the type of the property is string|undefined and the above
+    // translation handles it correctly.
+    // But in the special case of any, which includes undefined:
+    //   foo?: any;
+    // The type is just "any" (because that covers undefined as well) and
+    // our translation of the type is just "?".  But Closure wants it to instead
+    // be ?|undefined.
```

Comment:
Coming back to this code review, I *again* stumbled over this comment. Can you change `But Closure wants it to instead be ?|undefined.` to `But for properties (unlike e.g. variables) Closure wants it to instead be ?|undefined.`? I don't blame your comment writing, it's just that the behaviour is so strange.